plugin-dell/dphandler/storage.go

```diff
@@ -364,23 +350,67 @@ func compareCollection(list1, list2 []string) string {
 	return result
 }
 
-// createResponse is used for creating a final response for create volume
-func createResponse(code, msg, msgID string) []byte {
-	resp := dpresponse.ErrorResopnse{
-		Error: dpresponse.Error{
-			Code:    code,
-			Message: "See @Message.ExtendedInfo for more information.",
-			MessageExtendedInfo: []dpresponse.MsgExtendedInfo{
-				dpresponse.MsgExtendedInfo{
-					MessageID:   msgID,
-					Message:     msg,
-					MessageArgs: []string{},
+func updateTask(taskID, host, targetURI, request string, taskState dputilities.TaskState, taskStatus dputilities.TaskStatus,
+	percentComplete int32, httpMethod string) {
+	payLoad := &taskproto.Payload{
+		HTTPOperation: httpMethod,
+		JSONBody:      request,
+		TargetURI:     targetURI,
+	}
+
+	err := taskService.UpdateTask(taskID, host, taskState, taskStatus, percentComplete, payLoad, time.Now())
+	if err != nil {
+		log.Errorf("Unable to update task with ID: %s", taskID)
+	}
+}
+
+func retrieveTaskID(taskURI string) string {
+	strArray := strings.Split(taskURI, "/")
+	if strings.HasSuffix(taskURI, "/") {
+		return strArray[len(strArray)-2]
+	} else {
+		return strArray[len(strArray)-1]
+	}
+}
+
+func updateTaskWithException(taskID, host, uri, requestBody, method string) {
+	updateTask(taskID, host, uri, requestBody, dputilities.Exception, dputilities.Critical, 100, method)
+}
+
+func validateRequest(requestBody dpmodel.Volume) (bool, response.RPC) {
+	if requestBody.VolumeType == "" {
```

Comment:
VolumeType is not a mandatory parameter, any reason why its mandated in dell plugin?

Reply:
Removed VolumeType validation.